pkg/daemon/update_test.go

```diff
@@ -0,0 +1,62 @@
+package daemon
+
+import (
+	"fmt"
+	"testing"
+
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/openshift/machine-config-operator/pkg/generated/clientset/versioned/fake"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+)
+
+// TestUpdateOS verifies the return errors from attempting to update the OS follow expectations
+func TestUpdateOS(t *testing.T) {
+	// expectedError is the error we will use when expecting an error to return
+	expectedError := fmt.Errorf("broken")
+
+	// testClient is the NodeUpdaterClient mock instance that will front
+	// calls to update the host.
+	testClient := RpmOstreeClientMock{
+		GetBootedOSImageURLReturns: []GetBootedOSImageURLReturn{},
+		RunPivotReturns: []error{
+			// First run will return no error
+			nil,
+			// Second rrun will return our expected error
+			expectedError},
+	}
+
+	// Create a Daemon instance with mocked clients
+	d := Daemon{
```

Comment:
I would expect creating a test `Daemon` instance would likely be added as well to keep many of the tests from recreating the instances each time.